frontend/src/components/editor/Cell.tsx

```diff
@@ -311,10 +313,82 @@ const CellComponent = (
     [cellRef, editorView],
   );
 
+  const isCellCodeShown = !cellConfig.hide_code || temporarilyVisible;
+  const isMarkdown =
+    getCurrentLanguageAdapter(editorView.current) === "markdown";
+  const isMarkdownCodeHidden = isMarkdown && !isCellCodeShown;
+
+  const cellContainerRef = useRef<HTMLDivElement>(null);
+
+  // DOM node where the editorView will be mounted
+  const editorViewParentRef = useRef<HTMLDivElement>(null);
+
+  // if the cell is hidden, show the code editor temporarily
+  const temporarilyShowCode = useCallback(async () => {
+    if (!isCellCodeShown && setTemporarilyVisible !== undefined) {
+      setTemporarilyVisible(true);
+      editorView.current?.focus();
+      // Reach one parent up
+      const parent = editorViewParentRef.current?.parentElement;
+      const abortController = new AbortController();
+      parent?.addEventListener(
+        "focusout",
+        () => {
+          requestAnimationFrame(() => {
+            // Skip closing if the focus is still in the parent element
+            const focusedElement = document.activeElement;
+            if (parent?.contains(focusedElement)) {
+              return;
+            }
+            // Hide the code editor
+            setTemporarilyVisible(false);
+            editorView.current?.dom.blur();
+            abortController.abort();
+          });
+        },
+        { signal: abortController.signal },
+      );
+    }
+  }, [isCellCodeShown, setTemporarilyVisible, editorViewParentRef, editorView]);
```

Comment:
I'm not sure I understand your vision. 🤔

But imo, its complicated because of the various refs it uses. `editorViewParentRef` is passed to a child component.
`editorView` ref is in the current component. So, maybe a refactor would be in the composition of this component.

Reply:
Yea, its a bit of spaghetti (has been touched by 3 people), so maybe a rip/refactor is best. No need to do it in this PR if that is a distraction, but if it helps with this PR, feel free to.

Reply:
this PR's getting a bit scary, let me know if I should add more e2e tests? To ensure visual regression. Or work on this in pieces.

Also, do you know why the current e2e test is failing 😅? Works fine locally.

Reply:
by the way, is there a reason for the async eventListener handling? This seems to work

```.tsx
  const temporarilyShowCode = useCallback(() => {
    if (!isCellCodeShown) {
      setTemporarilyVisible(true);
      editorView.current?.focus();
      // Reach one parent up
      const parent = editorViewParentRef.current?.parentElement;

      const handleFocusOut = () => {
        requestAnimationFrame(() => {
          if (!parent?.contains(document.activeElement)) {
            // Hide the code editor
            setTemporarilyVisible(false);
            editorView.current?.dom.blur();
            parent?.removeEventListener("focusout", handleFocusOut);
          }
        });
      };
      parent?.addEventListener("focusout", handleFocusOut);
    }
  }, [isCellCodeShown, editorView]);
```